File: Geonil/week4/BOJ1260.py
import sys
from collections import defaultdict, deque

In = sys.stdin.readline
visited_dfs = []
visited_bfs = []


def bfs(graph, v_start):
    visited_bfs.append(v_start)
    queue = deque()
    queue.append(v_start)

    while True:
        if len(queue) == 0:
            break
        pop_v = queue.popleft()
        for v in graph[pop_v]:
            if v in visited_bfs:
                continue
            visited_bfs.append(v)
            queue.append(v)


def dfs(graph, v_start):
    visited_dfs.append(v_start)

    for vertex in graph[v_start]:
        if vertex not in visited_dfs:
            dfs(graph, vertex)


def main():
    v_num, e_num, v_start = map(int, In().split())
    graph = defaultdict(list)
    for _ in range(e_num):
        v1, v2 = map(int, In().split())
        graph[v1].append(v2)
        graph[v2].append(v1)

    for key in graph:
        graph[key].sort()

    dfs(graph, v_start)
    bfs(graph, v_start)

    for v in visited_dfs:
        print(v, end=' ')
    print()
    for v in visited_bfs:
        print(v, end=' ')


main()

File: Hanna/Week4/BOJ1260.c
/*
그래프를 DFS로 탐색한 결과와 BFS로 탐색한 결과를 출력하는 프로그램을 작성하시오.
단, 방문할 수 있는 정점이 여러 개인 경우에는 정점 번호가 작은 것을 먼저 방문하고,
더 이상 방문할 수 있는 점이 없는 경우 종료한다. 
정점 번호는 1번부터 N번까지이다.

첫째 줄에 정점의 개수 N(1 ≤ N ≤ 1,000), 간선의 개수 M(1 ≤ M ≤ 10,000), 
탐색을 시작할 정점의 번호 V가 주어진다. 
다음 M개의 줄에는 간선이 연결하는 두 정점의 번호가 주어진다.
어떤 두 정점 사이에 여러 개의 간선이 있을 수 있다. 
입력으로 주어지는 간선은 양방향이다.

5 5 3
5 4
5 2
1 2
3 4
3 1

3 1 4 2 5

4 5 1
1 2
1 3
1 4
2 4
3 4

1 2 3 4
*/

#include <stdio.h>

struct queue
{
    int spot;
    s int edge;
    int start;
};

int main()
{
    queue q[1000];
    int graph[1000][1000];

    scanf("%d %d %d", &spot, &line, &start);

    for (int i = 0; i < spot; i++)
    {
        for (int j = 0; j < line; j++)
        {
            scanf("%d %d", &);
        }
    }
    return 0;
}

int put(int spot)
{
    int queue[spot];
}